### DISCUSSION TITLE
Support for batch processing for Kafka consumer

### DISCUSSION THREAD

**Discussion by divyanshurai07** [apache/camel-karavan#736] [Q&A] [answered]
hi, i am trying to do batch processing of messages using kafka component. how can we process messages as a batch from kafka consumer?

Kafka component used : https://camel.apache.org/components/3.20.x/kafka-component.html#_configuring_component_options

**Comment by oscerd** [comment]
There is no support for batch consumer out of the box. You can use an aggregator eventually.

**Comment by davsclaus** [comment, marked as answer]
There is a ticket about adding batch support in camel-kafka
https://issues.apache.org/jira/browse/CAMEL-19241
